src/main/java/seedu/address/logic/commands/EditAppointmentCommand.java

```diff
@@ -0,0 +1,182 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_REMARK;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TIME;
+
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.appointment.Appointment;
+import seedu.address.model.appointment.Date;
+import seedu.address.model.appointment.Time;
+import seedu.address.model.name.Name;
+import seedu.address.model.remark.Remark;
+
+
+
+/**
+ * Edits an appointment in the app.
+ */
+public class EditAppointmentCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit appointment";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits an appointment in the app. \n"
+            + "Parameters: INDEX"
+            + PREFIX_NAME + "NAME "
+            + PREFIX_REMARK + "REMARK "
+            + PREFIX_DATE + "DATE "
+            + PREFIX_TIME + "TIME \n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "Meet Alex "
+            + PREFIX_REMARK + "at M hotel "
+            + PREFIX_DATE + "17-2-2021 "
+            + PREFIX_TIME + "1500 ";
+
+    public static final String MESSAGE_SUCCESS = "Edited appointment: %1$s";
+    public static final String MESSAGE_DUPLICATE_PROPERTY = "This appointment already exists in the app";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+
+    private final Index index;
+    private final EditAppointmentDescriptor editAppointmentDescriptor;
+
+    /**
+     * Creates an EditPropertyCommand to edit the specified {@code Property}.
+     */
```

Comment:
Rmb to change this javadoc

Reply:
Thanks for catching it! should all be fixed now